paimon-common/src/main/java/org/apache/paimon/format/FileFormat.java

```diff
@@ -77,18 +78,35 @@ public Optional<FileStatsExtractor> createStatsExtractor(RowType type) {
 
     /** Create a {@link FileFormat} from table options. */
     public static FileFormat fromTableOptions(
-            Options tableOptions, ConfigOption<String> formatOption) {
+            Options tableOptions,
+            ConfigOption<String> formatOption,
+            ConfigOption<Integer> readBatchSizeOption) {
```

Comment:
Maybe we can remove this method.
Provide a util static method in `CoreOptions`.